src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -981,249 +683,288 @@ public InputStream retrieveMetadata(String pid, String formatId)
+            releaseObjectLockedIds(pid);
         }
     }
 
+
     @Override
-    public void deleteObject(String pid) throws IllegalArgumentException, FileNotFoundException,
-        IOException, NoSuchAlgorithmException, InterruptedException,
-        PidNotFoundInCidRefsFileException {
-        logFileHashStore.debug(
-            "FileHashStore.deleteObject - Called to delete object for pid: " + pid
-        );
+    public void deleteInvalidObject(
+        ObjectMetadata objectInfo, String checksum, String checksumAlgorithm, long objSize)
+        throws NonMatchingObjSizeException, NonMatchingChecksumException,
+        UnsupportedHashAlgorithmException, InterruptedException, NoSuchAlgorithmException,
+        IOException {
+        logFileHashStore.debug("Verifying data object for cid: " + objectInfo.getCid());
         // Validate input parameters
-        FileHashStoreUtility.ensureNotNull(pid, "pid", "deleteObject");
-        FileHashStoreUtility.checkForEmptyString(pid, "pid", "deleteObject");
-
-        // First, find the object and evaluate its state
-        String cid;
-        try {
-            cid = findObject(pid);
-
-        } catch (OrphanPidRefsFileException oprfe) {
-            // Delete the pid refs file and return, nothing else to delete.
-            Path absPidRefsPath = getRealPath(pid, "refs", "pid");
-            Files.delete(absPidRefsPath);
+        FileHashStoreUtility.ensureNotNull(objectInfo, "objectInfo", "deleteInvalidObject");
+        FileHashStoreUtility.ensureNotNull(checksum, "checksum", "deleteInvalidObject");
+        FileHashStoreUtility.ensureNotNull(checksumAlgorithm, "checksumAlgorithm", "deleteInvalidObject");
+        FileHashStoreUtility.checkNotNegativeOrZero(objSize, "deleteInvalidObject");
 
-            String warnMsg = "FileHashStore.deleteObject - Cid refs file does not exist for pid: "
-                + pid + ". Deleted orphan pid refs file.";
-            logFileHashStore.warn(warnMsg);
-            return;
-
-        } catch (PidNotFoundInCidRefsFileException pnficrfe) {
-            // Delete pid refs file and return, nothing else to delete
-            Path absPidRefsPath = getRealPath(pid, "refs", "pid");
-            Files.delete(absPidRefsPath);
-
-            String warnMsg =
-                "FileHashStore.deleteObject - Pid not found in expected cid refs file for pid: "
-                    + pid + ". Deleted orphan pid refs file.";
-            logFileHashStore.warn(warnMsg);
-            return;
-        }
-
-        // If cid has been retrieved without any errors, proceed with second stage of deletion.
-        synchronized (referenceLockedCids) {
-            while (referenceLockedCids.contains(cid)) {
-                try {
-                    referenceLockedCids.wait(TIME_OUT_MILLISEC);
+        String objCid = objectInfo.getCid();
+        long objInfoRetrievedSize = objectInfo.getSize();
+        Map<String, String> hexDigests = objectInfo.getHexDigests();
```

Comment:
We should handle the case that `hexDigests` is null.

Reply:
Thank you, I have added in checks!